tests/translator/output/canaries_with_no_artifact_location.json

```diff
@@ -0,0 +1,124 @@
+{
+  "Resources": {
+    "CanaryWithDefinedArtifactLocation": {
+      "Type": "AWS::Synthetics::Canary",
+      "Properties": {
+        "ArtifactS3Location": "s3://hello",
+        "Code": {
+          "S3Bucket": "sam-bucket",
+          "S3Key": "hello.zip",
+          "Handler": "hello.handler"
+        },
+        "ExecutionRoleArn": "hello",
+        "Name": "canary1",
+        "RuntimeVersion": "python2.7",
+        "Schedule": {
+          "Expression": "rate(0 minutes)"
+        },
+        "StartCanaryAfterCreation": true,
+        "Tags": [
+          {
+            "Key": "lambda:createdBy",
+            "Value": "SAM"
+          }
+        ]
+      }
+    },
+    "FirstCanaryWithNoArtifactLocation": {
+      "Type": "AWS::Synthetics::Canary",
+      "Properties": {
+        "ArtifactS3Location": {
+          "Fn::Join": [
+            "",
+            [
+              "s3://",
+              {
+                "Ref": "FirstCanaryWithNoArtifactLocationArtifactBucket"
+              }
+            ]
+          ]
+        },
+        "Code": {
+          "Handler": "hello.handler",
+          "Script": "hello"
+        },
+        "ExecutionRoleArn": "hello",
+        "Name": "canary2",
+        "RuntimeVersion": "python2.7",
+        "Schedule": {
+          "Expression": "rate(0 minutes)"
+        },
+        "StartCanaryAfterCreation": true,
+        "Tags": [
+          {
+            "Key": "lambda:createdBy",
+            "Value": "SAM"
+          }
+        ]
+      }
+    },
+    "FirstCanaryWithNoArtifactLocationArtifactBucket": {
+      "Type": "AWS::S3::Bucket",
+      "DeletionPolicy": "Retain",
+      "Properties": {
+        "BucketEncryption": {
+          "ServerSideEncryptionConfiguration": [
+            {
+              "ServerSideEncryptionByDefault": {
+                "SSEAlgorithm": "AES256"
+              }
+            }
+          ]
+        }
+      }
+    },
+    "SecondCanaryWithNoArtifactLocation": {
+      "Type": "AWS::Synthetics::Canary",
+      "Properties": {
+        "ArtifactS3Location": {
+          "Fn::Join": [
+            "",
+            [
+              "s3://",
+              {
+                "Ref": "SecondCanaryWithNoArtifactLocationArtifactBucket"
+              }
+            ]
+          ]
+        },
+        "Code": {
+          "Handler": "hello.handler",
+          "Script": "hello"
+        },
+        "ExecutionRoleArn": "hello",
+        "Name": "canary3",
+        "RuntimeVersion": "python2.7",
+        "Schedule": {
+          "Expression": "rate(0 minutes)"
+        },
+        "StartCanaryAfterCreation": true,
+        "Tags": [
+          {
+            "Key": "lambda:createdBy",
+            "Value": "SAM"
+          }
+        ]
+      }
+    },
+    "SecondCanaryWithNoArtifactLocationArtifactBucket": {
+      "Type": "AWS::S3::Bucket",
+      "DeletionPolicy": "Retain",
```

Comment:
So this will create the bucket and will retain it after integration tests will clean up the Cloudformation stack. Can you ensure that this bucket gets deleted after you finish running integration tests? If not can you actually update integ tests to delete this bucket?

For more context the problem with leaving the bucket is that S3 have a 100 bucket limit per account. If your tests create 5 buckets per run, then we'll run out of limit in 20 or so runs of integration tests.

Reply:
yep, the bucket gets deleted once the stack is verified.